File: misc/benchmarks/python/cross_language_benchmark.py
def fib(n: int) -> int:
    if n == 0:
        return 0
    a = 0
    b = 1
    for _ in range(1, n + 1):
        nxt = a + b
        a = b
        b = nxt
    return a


def sum_squares(limit: int) -> int:
    total = 0
    for i in range(1, limit + 1):
        total += i * i
    return total


def prime_count(limit: int) -> int:
    count = 0
    for candidate in range(2, limit + 1):
        is_prime = True
        if candidate > 2:
            for divisor in range(2, candidate):
                if candidate % divisor == 0:
                    is_prime = False
                    break
        if is_prime:
            count += 1
    return count


def main() -> None:
    sum_part = sum_squares(2_000_000)
    fib_part = fib(35) * 2_000
    prime_part = prime_count(5_000)
    checksum = sum_part + fib_part + prime_part
    print(f"CHECKSUM:{checksum}")


if __name__ == "__main__":
    main()

File: misc/benchmarks/cpp/cross_language_benchmark.cpp
#include <cstdint>
#include <iostream>

static std::int64_t fib(int n) {
    if (n == 0) return 0;
    std::int64_t a = 0;
    std::int64_t b = 1;
    for (int i = 1; i <= n; ++i) {
        std::int64_t next = a + b;
        a = b;
        b = next;
    }
    return a;
}

static std::int64_t sumSquares(int limit) {
    std::int64_t total = 0;
    for (int i = 1; i <= limit; ++i) {
        total += static_cast<std::int64_t>(i) * static_cast<std::int64_t>(i);
    }
    return total;
}

static std::int64_t primeCount(int limit) {
    std::int64_t count = 0;
    for (int candidate = 2; candidate <= limit; ++candidate) {
        bool isPrime = true;
        if (candidate > 2) {
            for (int divisor = 2; divisor < candidate; ++divisor) {
                if (candidate % divisor == 0) {
                    isPrime = false;
                    break;
                }
            }
        }
        if (isPrime) {
            ++count;
        }
    }
    return count;
}

int main() {
    std::int64_t sumPart = sumSquares(2000000);
    std::int64_t fibPart = fib(35) * 2000;
    std::int64_t primePart = primeCount(5000);
    std::int64_t checksum = sumPart + fibPart + primePart;
    std::cout << "CHECKSUM:" << checksum << '\n';
    return 0;
}
